livedata/src/main/java/io/getstream/chat/android/livedata/service/sync/OfflineSyncFirebaseMessageService.kt

```diff
@@ -0,0 +1,147 @@
+package io.getstream.chat.android.livedata.service.sync
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.errors.ChatError
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.notifications.handler.ChatNotificationHandler
+import io.getstream.chat.android.client.socket.InitConnectionListener
+import io.getstream.chat.android.livedata.ChatDomain
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+class OfflineSyncFirebaseMessagingService : FirebaseMessagingService() {
+
+    private val logger = ChatLogger.get("OfflineSyncFirebaseMessagingService")
+
+    private val syncModule by lazy {
```

Comment:
Do we need this class?
It seems only a holder for `EncryptedBackgroundSyncConfigStore` and `NotificationConfigStore` that don't need anything else to be built, they could be here instead